dagsync/httpsync/sync.go

```diff
@@ -52,16 +52,21 @@ func NewSync(lsys ipld.LinkSystem, client *http.Client, blockHook func(peer.ID,
 }
 
 // NewSyncer creates a new Syncer to use for a single sync operation against a peer.
-func (s *Sync) NewSyncer(peerID peer.ID, peerAddr multiaddr.Multiaddr, rateLimiter *rate.Limiter) (*Syncer, error) {
-	rootURL, err := maconv.ToURL(peerAddr)
-	if err != nil {
-		return nil, err
+func (s *Sync) NewSyncer(peerID peer.ID, peerAddrs []multiaddr.Multiaddr, rateLimiter *rate.Limiter) (*Syncer, error) {
+	urls := make([]*url.URL, len(peerAddrs))
+	for i := range peerAddrs {
+		var err error
+		urls[i], err = maconv.ToURL(peerAddrs[i])
+		if err != nil {
+			return nil, err
+		}
```

Comment:
do we care about a publisher with both an HTTP and a graphsync multiaddr? if one of the addrs isn't http this will fail for all, as written

Reply:
If a publisher has _any_ HTTP addresses, then _only_ the HTTP addresses are selected and the sync is done with the HTTP publisher using the HTTP addresses.  So, an HTTP sync will never use a non-HTTP address.

Reply:
we should make sure that if someone sets an http address, and then reverts back, we don't remember the old http address and fail to revert back to graphsync pulls.

Reply:
So we will remember, and prefer the HTTP address, but if it is not being put back into the http peerstore by any new announce messages or manual syncs, then it will age out and we will go back to using whatever has been given to us since.